README.md

```diff
@@ -1,85 +1,123 @@
-of the general `fftokens` API specification.
+This will cause FireFly to parse the interface and provide ABI details
+to this connector, so it can determine the best methods from the ABI to be used for each operation.
+When this procedure is followed, the connector can find and call any variant of mint/burn/transfer/approval
+that is listed in the source code for [erc20.ts](src/tokens/erc20.ts) and [erc721.ts](src/tokens/erc721.ts).
+This list includes methods in the base standards, methods in the `IERC20WithData` and `IERC721WithData`
+interfaces defined in this repository, and common method variants from the
+[OpenZeppelin Wizard](https://wizard.openzeppelin.com). Additional variants can be added to the list
+by building a custom version of this connector or by proposing them via pull request.
 
-**POST /createpool**
+If implementing a new contract, the signatures in [IERC20WithData](samples/solidity/contracts/IERC20WithData.sol)
+and [IERC721WithData](samples/solidity/contracts/IERC721WithData.sol) will provide the most complete
+FireFly functionality by allowing FireFly transactions and messages to be pinned to the blockchain
+alongside token operations. The sample [ERC20WithData](samples/solidity/contracts/ERC20WithData.sol)
+and [ERC721WithData](samples/solidity/contracts/ERC721WithData.sol) contracts may be used to
+get up and running with simple token support, and may provide a starting point for developing
+production contracts that can be used with this connector.
 
-```
-config: {
-  address?: string, // Address of a pre-deployed contract to index
-  blockNumber?: string, // Starting block to begin indexing (only valid with "address")
-}
-```
+### Solidity Interface Support
 
-**POST /approval**
+In the absence of being provided with ABI details, the token connector will attempt to guess the contract
+ABI in use. It does this by using ERC165 `supportsInterface()` to query the contract's support for `IERC20WithData`
+or `IERC721WithData`, as defined in this repository. If the query succeeds, the connector will leverage
+the methods on that interface to perform token operations. Therefore it is possible to use these
+contracts without the extra step of teaching FireFly about the contract interface first.
 
-```
-config: {
-  allowance?: string, // Number of tokens operator is allowed to manage (only valid with ERC20)
-  tokenIndex?: string, // A specific NFT operator is allowed to manage (only valid with ERC721)
-}
-```
+### Fallback Functionality (not recommended)
+
+If neither of the above procedures is followed for a given contract, the connector will fall back to assuming
+that the ABI looks like [ERC20NoData.json](src/abi/ERC20NoData.json) or
+[ERC721NoData.json](src/abi/ERC721NoData.json), which are based on common OpenZeppelin patterns. This
+behavior can also be tweaked to assume [ERC20NoDataOld.json](src/abi/ERC20NoDataOld.json) or
```

Comment:
Note: I changed some of these filenames in #109, so depending on merge order, they may need to be adjusted here.